inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -737,6 +751,80 @@ void InsertCopyLayerPass::run() {
     }
 }
 
+void Concat4Dto2DPass::run() {
+    auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(pLayers->front());
+
+    if (getPassManager()->getPolicy().ConcatConversionPolicy == Policy::Concat4Dto2DConversion::DISABLED) return;
+    if (getPassManager()->getPolicy().ConcatAlignmentPolicy == Policy::ConcatAlignment::DISABLED) return;
+    if (getPassManager()->getPolicy().ConcatAlignmentPolicy == Policy::ConcatAlignment::DISABLED_FOR_FP32 && !quantized) return;
+
+    for (auto & l : *pLayers) {
+        LayerInfo info(l);
+        if (!info.isConcat()) continue;
+        auto concatLayer = info.as<ConcatLayer*>();
+
+        auto dims_size = concatLayer->insData[0].lock()->getDims().size();
```

Comment:
I know it is not expected, but it is still good to check if there is at least 1 input to concat